Raven.Client.Lightweight/Connection/ReplicationInformer.cs

```diff
@@ -72,20 +79,22 @@ private Task UpdateReplicationInformationIfNeededInternalAsync(string url, Func<
                 firstTime = false;
 
                 if (lastReplicationUpdate.Add(Conventions.TimeToWaitBetweenReplicationTopologyUpdates) > SystemTime.UtcNow && force == false)
-                    return new CompletedTask();
+                    return;
 
-                var taskCopy = refreshReplicationInformationTask;
-                if (taskCopy != null)
-                    return taskCopy;
+                if (refreshReplicationInformationTask != null)
+                    await refreshReplicationInformationTask.ConfigureAwait(false);
 
-                return refreshReplicationInformationTask = Task.Factory.StartNew(() => 
-                    RefreshReplicationInformationInternal(url, getReplicationDestinations)).
+                refreshReplicationInformationTask = RefreshReplicationInformationInternalAsync(url, getReplicationDestinations).
```

Comment:
If I understood the code correctly the rationale is to do the heavy lifting in the background but keep track of it so I kept that consistent

Reply:
Yes, looks okay